src/pytorch_lightning/utilities/types.py

```diff
@@ -139,3 +139,25 @@ class LRSchedulerConfig:
     strict: bool = True
     # opt_idx assigned internally if not assigned by user
     opt_idx: Optional[int] = None
+
+
+class _SentinelMeta(type):
+    """Metaclass representing a sentinel value by the name of the class.
+
+    Reference: https://stackoverflow.com/a/69243488/1162383
+    See also: https://peps.python.org/pep-0661/
+    """
+
+    def __repr__(cls) -> str:
+        return f"<{cls.__name__}>"
+
+    def __bool__(cls) -> Literal[False]:
+        return False
+
+
+class Sentinel(metaclass=_SentinelMeta):
+    """Subclass this to create a new sentinel."""
+
+
+class _NO_DATA(Sentinel):
+    """A sentinel representing the default value for 'no dataloader passed to Trainer method'."""
```

Comment:
Note: Sometimes we see sentinels defined as 

DEFAULT = object()

But this does not work for us, because pickling (e.g. ddp_spawn) will re-instantiate this object and then checks like `dataloader is not DEFAULT` would fail!